### PR TITLE
Glossary support

### PR DESCRIPTION
**Pull request by Josue-T**
# Jira URL

# Changes

## Clarifications

# Screenshots & Video

# Executed Tests

# Expected merging strategy

* Prefers squash: Yes 
* Backport on branches:
  *

### CODE REVIEW

File: translator-api/src/main/java/org/xwiki/contrib/translator/TranslatorScriptService.java [hunk 1]
```diff
@@ -156,4 +159,34 @@ public DocumentReference computeTranslationReference(DocumentReference originalD
         }
         return translator.computeTranslationReference(originalDocument, translationTitle, translationLocale);
     }
+
+    public List<GlossaryInfo> getGlossaries() throws TranslatorException
+    {
+        Translator translator = translatorManager.getTranslator();
+        if (this.authorizationManager.hasAccess(Right.PROGRAM)) {
```
Comment (slauriere) [inline translator-api/src/main/java/org/xwiki/contrib/translator/TranslatorScriptService.java]: Why do we need the PROGRAM right here?

File: translator-deepl/src/main/java/org/xwiki/contrib/translator/deepl/DeeplTranslator.java [hunk 2]
```diff
@@ -56,6 +91,10 @@ public String translate(String content, Locale from, Locale to, boolean html) th
         if (html) {
             options.setTagHandling("html");
         }
+        Optional<String> glossaryId = getGlossaryIdForLocales(from, to);
+        if (glossaryId.isPresent()){
```
Comment (slauriere) [inline translator-deepl/src/main/java/org/xwiki/contrib/translator/deepl/DeeplTranslator.java]: Missing space before the bracket -> was the XWiki Java code style correctly  applied?

File: translator-deepl/src/main/java/org/xwiki/contrib/translator/deepl/DeeplTranslator.java [hunk 3]
```diff
@@ -108,6 +147,94 @@ public Usage getUsage() throws DeepLException, InterruptedException
...
+            logger.debug("Error when getting glossaries [{}]", e.getMessage(), e);
+            throw new TranslatorException("Interrupt exception when getting glossaries", e);
+        } catch (DeepLException e) {
+            logger.debug("Error when getting glossaries [{}]", e.getMessage(), e);
+            throw new TranslatorException("DeepL exception when getting glossaries", e);
+        }
+    }
+
+    @Override
+    public Map<String, String> getGlossaryEntryDetails(String id) throws TranslatorException
```
Comment (slauriere) [inline translator-deepl/src/main/java/org/xwiki/contrib/translator/deepl/DeeplTranslator.java]: Might be renamed into `getGlossaryEntries` instead?

File: translator-api/src/main/java/org/xwiki/contrib/translator/TranslatorScriptService.java (hunk 1 above)
Comment (Josue-T) [reply translator-api/src/main/java/org/xwiki/contrib/translator/TranslatorScriptService.java]: Well, if we consider that all field into `GlossaryInfo` is not critical we can remove the check  right.

File: translator-deepl/src/main/java/org/xwiki/contrib/translator/deepl/DeeplTranslator.java (hunk 3 above)
Comment (Josue-T) [reply translator-deepl/src/main/java/org/xwiki/contrib/translator/deepl/DeeplTranslator.java]: Fixed by 66e6e01a77520d038148890cec8c57980b121f65

**Pull request merged**